Core/Core/DocViewer/DocViewerViewController.swift

```diff
@@ -101,7 +101,8 @@ public class DocViewerViewController: UIViewController {
         let document = Document(url: localURL)
         if let annotationMeta = metadata.annotations {
             document.defaultAnnotationUsername = annotationMeta.user_name
-            document.didCreateDocumentProviderBlock = { documentProvider in
+            document.didCreateDocumentProviderBlock = { [weak self] documentProvider in
```

Comment:
And this fixes a memory leak I spotted while trying to track down the original crash.